Manage the creation of SCM auxvariables, depending on mesh (#32518)
idaholab/moose: modules/subchannel/doc/content/modules/subchannel/general/SCMAuxVariables.md

- Enthalpy: "h" $J/kg$

- Temperature: "T" $K$

- Density: "rho" $kg/m^3$

- Viscosity: "mu" $Pa*s$

- Linear heat rate: "q_prime" $W/m$

- Subchannel centroid displacement: "displacement" $m$

An example usage of displacement can be found [here](areva_fctf.md).

### Flags

Enabling the boolean flag [!param](/Problem/QuadSubChannel1PhaseProblem/full_output) in the `[SubChannel]` problem creates the additional auxvariables:

- Pressure drop: "DP" $Pa$ (This refers to the cell value immediately below the node)

- Friction factor: "ff" $unitless$ (This refers to the cell value immediately below the node)

!alert note
This `DP` auxvariable variable is not calculated for the monolithic solve, in which case it will read zero in all nodes.

## Fuel pins mesh

This is the fuel pin mesh that gets created by: [SCMQuadPinMeshGenerator](SCMQuadPinMeshGenerator.md) and [SCMTriPinMeshGenerator](SCMTriPinMeshGenerator.md) for quadrilateral and triangular lattice respectively.

When the `[SubChannel]` problem syntax is present, defining either of these mesh generators in the fuel pin mesh block, automatically defines the following auxiliary variables that live on the fuel pin mesh nodes:

> - Fuel pin average surface temperature: $K$ = "Tpin"

Reviewer: ```suggestion
- Fuel pin average surface temperature: "Tpin"  $K$ 
```

## Duct mesh

This is the duct mesh that gets created by: [SCMQuadDuctMeshGenerator](SCMQuadDuctMeshGenerator.md) and [SCMTriDuctMeshGenerator](SCMTriDuctMeshGenerator.md) for square and triangular assemblies respetively.

When the `[SubChannel]` problem syntax is present, defining either of these mesh generators in the duct mesh block, automatically defines the following auxiliary variables that live on the duct mesh nodes:

- Duct heat flux: "duct_heat_flux" $W/m^2$

- Duct temperature: "Tduct" $K$

!alert warning
 All ICs, AuxKernels, and Postprocessors using these variables must be applied on compatible blocks. Mismatched block names will result in an error.

> !alert note

Reviewer: ```suggestion
## Auxiliary variables, block restrictions and output

```

# SCMAuxVariables

Depending on the mesh type defined in the SCM input file, SCM automatically creates and populates the following auxiliary variables on the mesh. These variables are defined at equally spaced nodes from the inlet to the outlet. A subchannel cell spans the distance between two consecutive nodes.

## SubChannel mesh

This is the Subchannel mesh that gets created by: [SCMQuadSubChannelMeshGenerator](SCMQuadSubChannelMeshGenerator.md) and [SCMTriSubChannelMeshGenerator](SCMTriSubChannelMeshGenerator.md) for quadrilateral and triangular lattice respectively.

When the `[SubChannel]` problem syntax is present, defining either of these mesh generators in the subchannel mesh block, automatically defines the following auxiliary variables that live on the subchannel mesh nodes:

- Mass flow rate: "mdot" $kg/s$

- Subchannel surface area: "S" $m^2$

- Subchannel wetted perimeter: "w_perim" $m$

- Net crossflow per subchannel cell: "SumWij" $kg/s$ (This refers to the cell value immediately below the node)

- Pressure: "P" $Pa$

- Enthalpy: "h" $J/kg$

- Temperature: "T" $K$

- Density: "rho" $kg/m^3$

- Viscosity: "mu" $Pa*s$

> - Linear heat rate: "q_prime" $W/m$

Reviewer: is this the nodal q_prime too?

Reviewer: or for the cell below

Author: nodal.

# SCMAuxVariables

Depending on the mesh type defined in the SCM input file, SCM automatically creates and populates the following auxiliary variables on the mesh. These variables are defined at equally spaced nodes from the inlet to the outlet. A subchannel cell spans the distance between two consecutive nodes.

## SubChannel mesh

This is the Subchannel mesh that gets created by: [SCMQuadSubChannelMeshGenerator](SCMQuadSubChannelMeshGenerator.md) and [SCMTriSubChannelMeshGenerator](SCMTriSubChannelMeshGenerator.md) for quadrilateral and triangular lattice respectively.

> When the `[SubChannel]` problem syntax is present, defining either of these mesh generators in the `[sub_channel]` mesh block, automatically defines the following auxiliary variables:

Reviewer: this is not true. 
The syntax is `[Subchannel / a name]` the "sub_channel" is just a name

> Mass flow rate: $kg/s$ = "mdot"

Reviewer: ```suggestion
- Mass flow rate: $kg/s$ = "mdot"
```
use the list it will look better than
https://mooseframework.inl.gov/docs/PRs/32518/site/modules/subchannel/general/SCMAuxVariables.html

> Subchannel surface area: $m^2$ = "S"

Reviewer: ```suggestion
Subchannel surface area: "S" $[m^2]$
```
this looks more regular

Subchannel wetted perimeter: $m$ = "w_perim"

> Net crossflow per subchannel cell: $kg/s$ = "SumWij" (This applies to the cell immediately behind the node)

Reviewer: this isnt clear
```suggestion
Net crossflow per subchannel cell: $kg/s$ = "SumWij" (the node value applies to the cell below the node on the Z axis)
```

Pressure: $Pa$ = "P"

Enthalpy: $J/kg$ = "h"

Temperature: $K$ = "T"

Density: $kg/m^3$ = "rho"

Viscosity: $Pa*s$ = "mu"

Linear heat rate: $W/m$ = "q_prime"

!alert note

> The subchannel mesh block needs to be named: `[sub_channel]`.

Reviewer: I dont see why in the changes you included

Reviewer: I m very much against this. We're looking to expand to multiple assemblies not hardcoding "sub_channel" as the single block here

Reviewer 2: I was trying to make the auxvariables block restricted.  But i figured a way to remove the hard coding of mesh names.

### Flags

- Enabling the boolean flag [!param](/Problem/QuadSubChannel1PhaseProblem/full_output) in the `[SubChannel]` problem creates the additional auxvariables:

Pressure drop: $Pa$ = "DP" (This applies to the cell immediately behind the node)

!alert note
This `DP` auxvariable variable is not calculated for the monolithic solve, in which case it will read zero in all nodes.

Friction factor: $unitless$ = "ff" (This applies to the cell immediately behind the node)

- Enabling the boolean flag [!param](/Problem/QuadSubChannel1PhaseProblem/deformation) in the `[SubChannel]` problem creates the additional auxvariables:

Subchannel centroid displacement: $m$ = "displacement"

An example usage of displacement can be found [here](areva_fctf.md).

## Fuel pins mesh

> This is the Pin mesh that gets created by: [SCMQuadPinMeshGenerator](SCMQuadPinMeshGenerator.md) and [SCMTriPinMeshGenerator](SCMTriPinMeshGenerator.md) for quadrilateral and triangular lattice respectively.

Reviewer: ```suggestion
This is the fuel pin mesh that gets created by: [SCMQuadPinMeshGenerator](SCMQuadPinMeshGenerator.md) and [SCMTriPinMeshGenerator](SCMTriPinMeshGenerator.md) for quadrilateral and triangular lattice respectively.
```

When the `[SubChannel]` problem syntax is present, defining either of these mesh generators in the `[fuel_pins]` mesh block, automatically defines the following auxiliary variables:

> Fuel pin average surface temperature: $K$ = "Tpin"

Reviewer: same comments

Fuel pin diameter: $m$ = "Dpin"

Linear heat rate: $W/m$ = "q_prime"

The variable `q_prime` is created on:

- the `fuel_pins` mesh if a pin mesh is present

- otherwise on the `sub_channel` mesh

Average subchannel convective heat transfer coefficient: $W/m^2K$ = "HTC"

!alert note
The fuel pin mesh block needs to be named: `[fuel_pins]`.

## Duct mesh

> This is the Duct mesh that gets created by: [SCMQuadDuctMeshGenerator](SCMQuadDuctMeshGenerator.md) and [SCMTriDuctMeshGenerator](SCMTriDuctMeshGenerator.md) for square and triangular assemblies respetively.

Reviewer: ```suggestion
This is the duct mesh that gets created by: [SCMQuadDuctMeshGenerator](SCMQuadDuctMeshGenerator.md) and [SCMTriDuctMeshGenerator](SCMTriDuctMeshGenerator.md) for square and triangular assemblies respectively.
```

When the `[SubChannel]` problem syntax is present, defining either of these mesh generators in the `[duct]` mesh block, automatically defines the following auxiliary variables:

Duct heat flux: $W/m^2$ = "duct_heat_flux"

Duct temperature: $K$ = "Tduct"

!alert note
The duct mesh block needs to be named: `[duct]`.

!alert warning

> Block names must match those defined in the mesh. All ICs, AuxKernels, and Postprocessors using these variables must be applied on compatible blocks. Mismatched block names will result in an error.

Reviewer: no keep the names of the fuel, duct and channels in the problem and set them appropriately in ICs
